.github/workflows/pr.yml

```diff
@@ -0,0 +1,19 @@
+name: "Pull Request Checks"
+on:
+  # Allow triggering runs on arbitrary branches
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+      - dev
+
+jobs:
+  run_tests:
+    name: Run Tests
+    runs-on: ubuntu-latest
```

Comment:
Note that `runs-on` is for the _virtual environment_ (OS of the runner of this workflow, analogous to your OS, when running locally), and that the resulting image should still respect any base images as specified in the `Dockerfile`.

That said, Ubuntu 18.04 is among the [currently supported runner OSes](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idruns-on), so this is possible, yes. The only thing about pinning to a specific version is that it will eventually be deprecated and need upgraded. If this year's deprecation of 16.04 is any indicator, then the GitHub Runner deprecation would happen +/- 5 months after the underlying Ubuntu version switches to ESM (so, somewhere around the end of 2023, if I [understand correctly](https://ubuntu.com/about/release-cycle)).

Would you still like me to switch to `ubuntu-18.04`?

Reply:
Ah okay. No, it seems wiser to leave as is in that case. Thanks for clarifying